ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/v5/DiscoveryV5Service.java

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright contributors to Hyperledger Besu.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.p2p.discovery.v5;
+
+import org.hyperledger.besu.crypto.KeyPair;
+
+import java.lang.reflect.Method;
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.apache.tuweni.crypto.SECP256K1;
+import org.ethereum.beacon.discovery.AddressAccessPolicy;
+import org.ethereum.beacon.discovery.DiscoverySystemBuilder;
+import org.ethereum.beacon.discovery.MutableDiscoverySystem;
+import org.ethereum.beacon.discovery.schema.NodeRecord;
+import org.ethereum.beacon.discovery.schema.NodeRecordBuilder;
+import org.ethereum.beacon.discovery.schema.NodeRecordFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Thin wrapper around DiscV5 for Besu using MutableDiscoverySystem (no Teku deps). */
+public final class DiscoveryV5Service implements AutoCloseable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DiscoveryV5Service.class);
+  private final MutableDiscoverySystem discovery;
+  private volatile boolean started = false;
+
+  // Fallback poller (used only if one or both callback methods are missing)
+  private final ScheduledExecutorService scheduler =
+      Executors.newSingleThreadScheduledExecutor(
+          r -> {
+            final Thread t = new Thread(r, "besu-discv5-poller");
+            t.setDaemon(true);
+            return t;
+          });
```

Comment:
Instead of creating a new scheduler, could be possible to use `EthScheduler::scheduleFutureTaskWithFixedDelay`